server/src/http.rs

```diff
@@ -397,7 +424,22 @@ impl Service {
             .and(warp::query::<PutParams>())
             .and(warp::body::bytes())
             .and(self.with_proxy())
-            .and_then(|ctx, params, points, proxy: Arc<Proxy>| async move {
+            .and(header::optional::<String>(CONTENT_ENCODING_HEADER))
+            .and_then(|ctx, params, points: Bytes, proxy: Arc<Proxy>, encoding: Option<String>| async move {
+                let points = match encoding {
+                    Some(encoding) => {
+                        let encode_type = EncodingType::try_from(&encoding[..])?;
+                        match encode_type {
+                            EncodingType::Gzip => {
+                                let mut d = GzDecoder::new(points.as_bytes());
+                                let mut decompressed_data = Vec::new();
```

Comment:
Give it a initial capacity, such as `points.as_bytes().len() * 2`, assuming the compression rate is 2.